pallets/ah-migrator/src/lib.rs

```diff
@@ -0,0 +1,1373 @@
+		#[pallet::call_index(102)]
+		#[pallet::weight(T::AhWeightInfo::set_dmp_queue_priority())]
+		pub fn set_dmp_queue_priority(
+			origin: OriginFor<T>,
+			new: DmpQueuePriority<BlockNumberFor<T>>,
+		) -> DispatchResult {
+			Self::ensure_admin_or_manager(origin)?;
+
+			let old = DmpQueuePriorityConfig::<T>::get();
+			if old == new {
+				return Err(Error::<T>::DmpQueuePriorityAlreadySet.into());
+			}
+			ensure!(
+				new.get_priority_blocks().map_or(true, |blocks| !blocks.is_zero()),
+				Error::<T>::InvalidParameter
+			);
+			DmpQueuePriorityConfig::<T>::put(new.clone());
+			Self::deposit_event(Event::DmpQueuePriorityConfigSet { old, new });
+			Ok(())
+		}
+
+		/// Set the manager account id.
+		///
+		/// The manager has the similar to [`Config::AdminOrigin`] privileges except that it
+		/// can not set the manager account id via `set_manager` call.
+		#[pallet::call_index(103)]
+		#[pallet::weight(T::AhWeightInfo::set_manager())]
+		pub fn set_manager(origin: OriginFor<T>, new: Option<T::AccountId>) -> DispatchResult {
+			<T as Config>::AdminOrigin::ensure_origin(origin)?;
+			let old = Manager::<T>::get();
+			Manager::<T>::set(new.clone());
+			Self::deposit_event(Event::ManagerSet { old, new });
+			Ok(())
+		}
+
+		/// Finish the migration.
+		///
+		/// This is typically called by the Relay Chain to signal the migration has finished.
+		///
+		/// The `data` parameter might be `None` if we are running the migration for a second time
+		/// for some pallets and have already performed the checking account balance correction,
+		/// so we do not need to do it this time.
+		#[pallet::call_index(110)]
+		#[pallet::weight(T::AhWeightInfo::finish_migration())]
+		pub fn finish_migration(
+			origin: OriginFor<T>,
+			data: Option<MigrationFinishedData<T::Balance>>,
+		) -> DispatchResult {
+			Self::ensure_admin_or_manager(origin)?;
+
+			Self::migration_finish_hook(data).map_err(Into::into)
+		}
+
+		/// XCM send call identical to the [`pallet_xcm::Pallet::send`] call but with the
+		/// [Config::SendXcm] router which will be able to send messages to the Asset Hub during
+		/// the migration.
+		#[pallet::call_index(111)]
+		#[pallet::weight({ Weight::from_parts(10_000_000, 1000) })]
+		pub fn send_xcm_message(
+			origin: OriginFor<T>,
+			dest: Box<VersionedLocation>,
+			message: Box<VersionedXcm<()>>,
+		) -> DispatchResult {
+			Self::ensure_admin_or_manager(origin.clone())?;
+
+			let origin_location = <T as pallet_xcm::Config>::SendXcmOrigin::ensure_origin(origin)?;
+			let interior: Junctions =
+				origin_location.clone().try_into().map_err(|_| Error::<T>::InvalidOrigin)?;
+			let dest = Location::try_from(*dest).map_err(|()| Error::<T>::BadXcmVersion)?;
+			let mut message: Xcm<()> =
+				(*message).try_into().map_err(|()| Error::<T>::BadXcmVersion)?;
+
+			if interior != Junctions::Here {
+				message.0.insert(0, DescendOrigin(interior.clone()));
+			}
+
+			// validate
+			let (ticket, _price) =
```

Comment:
It think `xcm::send::<Router>` will do the same as here: `validate_send`, and then `deliver`. Maybe you are doing it like this to just have better logging?